ocaml/xapi/xapi_vm.ml

```diff
@@ -60,9 +60,17 @@ let assert_can_boot_here ~__context ~self ~host =
   assert_can_boot_here ~__context ~self ~host ~snapshot ~do_cpuid_check:true ()
 
 let retrieve_wlb_recommendations ~__context ~vm =
+  let compat_string_of_float f =
+    let repr = string_of_float f in
+    (* Original format has always at least one fractional decimal digit *)
+    if Astring.String.is_suffix ~affix:"." repr then
```

Comment:
I assume performance is not an issue. Otherwise: `str.[String.length str - 1] = '.'` could be a cheaper predicate. In general, Atring is good about performance so I would not worry about this.

Reply:
I read astring's source code around comparisons and tends to do the minimal amount of work needed, I don't think we need to worry that much about a function that's rarely called.